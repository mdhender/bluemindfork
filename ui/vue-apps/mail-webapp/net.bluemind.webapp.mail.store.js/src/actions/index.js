export * from "./$_createFolder";
export * from "./$_getIfNotPresent";
export * from "./$_move";
export * from "./addAttachment";
export * from "./bootstrap";
export * from "./collapseFolder";
export * from "./deleteDraft";
export * from "./expandFolder";
export * from "./loadRange";
export * from "./loadUnreadCount";
export * from "./markAsRead";
export * from "./markAsUnread";
export * from "./move";
export * from "./remove";
export * from "./purge";
export * from "./saveDraft";
export * from "./search";
export * from "./selectFolder";
export * from "./selectMessage";
export * from "./send";
