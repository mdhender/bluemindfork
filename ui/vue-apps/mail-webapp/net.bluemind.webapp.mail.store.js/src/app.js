export const state = {
    activeFolder: undefined
};

export const mutations = {
    SET_ACTIVE_FOLDER: (state, key) => {
        state.activeFolder = key;
    }
};

export const getters = {
    CURRENT_MAILBOX: state => state.mailboxes[state.folders[state.activeFolder].mailbox]
};
