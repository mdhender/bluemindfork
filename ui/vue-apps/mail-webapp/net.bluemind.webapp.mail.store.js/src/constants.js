export const STATUS = {
    IDLE: "idle",
    RESOLVED: "resolved",
    REJECTED: "rejected",
    LOADING: "loading"
};
