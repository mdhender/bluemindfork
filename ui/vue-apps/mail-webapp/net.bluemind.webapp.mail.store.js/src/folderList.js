export const TOGGLE_EDIT_FOLDER = "TOGGLE_EDIT_FOLDER";

export const state = {
    folderList: {
        editing: undefined
    }
};

export const mutations = {
    [TOGGLE_EDIT_FOLDER]: (state, key) => {
        console.log("COUCOUUUUU TOGGLE EDIT FOLDER !!!!!!!!", key);
        if (state.folderList.editing && state.folderList.editing === key) {
            state.folderList.editing = undefined;
        } else {
            state.folderList.editing = key;
        }
    }
};
