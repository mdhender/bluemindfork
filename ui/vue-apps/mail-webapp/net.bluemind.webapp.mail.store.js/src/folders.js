import Vue from "vue";
import { FolderAdaptor } from "./helpers/FolderAdaptor";
import { inject } from "@bluemind/inject";
import { ItemFlag } from "@bluemind/core.container.api";

export const ADD_FOLDER = "ADD_FOLDER";
export const ADD_FOLDERS = "ADD_FOLDERS";
export const RENAME_FOLDER = "RENAME_FOLDER";
export const REMOVE_FOLDER = "REMOVE_FOLDER";
export const SET_UNREAD_COUNT = "SET_UNREAD_COUNT";
export const TOGGLE_FOLDER = "TOGGLE_FOLDER";

export const FETCH_FOLDERS = "FETCH_FOLDERS";
export const CREATE_FOLDER = "CREATE_FOLDER";

export const state = {
    folders: {}
};

export const getters = {
    ["HAS_CHILDREN_GETTER"]: state => key => !!Object.values(state.folders).find(folder => folder.parent === key),
    ["MAILSHARE_FOLDERS"]: (state, getters) => {
        return Object.values(state.folders)
            .filter(folder => getters["MAILSHARE_KEYS"].includes(folder.mailbox))
            .map(folder => folder.key);
    },
    ["MY_MAILBOX_FOLDERS"]: (state, getters) =>
        Object.values(state.folders)
            .filter(folder => getters["MY_MAILBOX_KEY"] === folder.mailbox)
            .map(folder => folder.key)
};

export const mutations = {
    [CREATE_FOLDER]: (state, { key, name, parent, mailbox }) => {
        const folder = FolderAdaptor.create(key, name, parent && state.folders[parent], mailbox);
        Vue.set(state.folders, folder.key, folder);
    },
    [ADD_FOLDER]: (state, folder) => {
        Vue.set(state.folders, folder.key, folder);
    },
    [ADD_FOLDERS]: (state, folders) => folders.forEach(folder => Vue.set(state.folders, folder.key, folder)),
    [RENAME_FOLDER]: (state, { key, name }) => {
        Vue.set(state.folders, key, FolderAdaptor.rename(state.folders[key], name));
    },
    [REMOVE_FOLDER]: (state, key) => {
        Vue.delete(state.folders, key);
    },
    [SET_UNREAD_COUNT]: (state, { key, count }) => {
        if (count >= 0 && state.folders[key].unread !== count) {
            state.folders[key].unread = count;
        }
    },
    [TOGGLE_FOLDER]: (state, key) => {
        state.folders[key].expanded = !state.folders[key].expanded;
    }
};

export const actions = {
    async [FETCH_FOLDERS]({ commit }, mailbox) {
        const items = await inject("MailboxFoldersPersistence", mailbox.uid).all();
        const folders = items
            .filter(item => !item.flags.includes(ItemFlag.Deleted))
            .sort((a, b) => a.value.fullName.toLowerCase().localeCompare(b.value.fullName.toLowerCase()))
            .map(item => FolderAdaptor.fromMailboxFolder(item, mailbox));
        commit(ADD_FOLDERS, folders);
    },

    async [CREATE_FOLDER]({ commit, state }, { key, name, parent, mailbox }) {
        commit(CREATE_FOLDER, { key, name, parent, mailbox });
        const folder = state.folders[key];
        const item = FolderAdaptor.toMailboxFolder(folder, state.folders[folder.parent], mailbox);
        try {
            const { uid, id } = await inject("MailboxFoldersPersistence", mailbox.uid).createBasic(item.value);
            commit(ADD_FOLDER, { ...folder, uid, id });
        } catch (e) {
            commit(REMOVE_FOLDER, key);
            throw e;
        }
    },
    //FIXME : when deleting a folder having children, it is not deleted in UI & we got errors in the console..
    async [REMOVE_FOLDER]({ state, commit }, { key, mailbox }) {
        const folder = state.folders[key];
        commit(REMOVE_FOLDER, key);
        try {
            const service = inject("MailboxFoldersPersistence", mailbox.uid);
            await service.deepDelete(folder.id);
        } catch (e) {
            commit(ADD_FOLDER, folder);
            throw e;
        }
    },
    async [RENAME_FOLDER]({ commit, state }, { name, key, mailbox }) {
        const oldName = state.folders[key].name;
        commit(RENAME_FOLDER, { name, key, mailbox });
        const folder = state.folders[key];
        const item = FolderAdaptor.toMailboxFolder(folder, state.folders[folder.parent], mailbox);
        try {
            await inject("MailboxFoldersPersistence", mailbox.uid).updateById(item.internalId, item.value);
        } catch (e) {
            commit(RENAME_FOLDER, { name: oldName, key, mailbox });
            throw e;
        }
    }
};
