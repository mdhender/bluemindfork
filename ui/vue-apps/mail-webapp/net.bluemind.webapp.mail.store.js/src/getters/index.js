export * from "./currentFolder";
export * from "./currentMailbox";
export * from "./lastRecipients";
export * from "./mailshares";
export * from "./my";
export * from "./nextMessageKey";
export * from "./tree";
export * from "./unreadCount";
export * from "./areMessagesFiltered";
export * from "./messageListMode";
export * from "./status";
