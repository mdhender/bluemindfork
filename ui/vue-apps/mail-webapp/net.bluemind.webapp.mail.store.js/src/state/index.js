import { STATUS } from "../constants";

export let currentFolderKey;
export let messageFilter;
export let foldersData = {};
export let login;
export let maxMessageSize;
export const status = STATUS.IDLE;
