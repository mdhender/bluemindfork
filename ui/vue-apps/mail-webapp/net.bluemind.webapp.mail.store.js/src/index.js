export { FolderAdaptor } from "./helpers/FolderAdaptor";
export { MailboxAdaptor } from "./helpers/MailboxAdaptor";
import * as folders from "./folders";
import * as folderList from "./folderList";
import * as mailboxes from "./mailboxes";

export default {
    namespaced: true,
    state: { ...folders.state, ...folderList.state, ...mailboxes.state },
    actions: { ...folders.actions, ...mailboxes.actions },
    mutations: { ...folders.mutations, ...folderList.mutations, ...mailboxes.mutations }
};

export const ADD_FOLDER = "mail/" + folders.ADD_FOLDER;
export const ADD_FOLDERS = "mail/" + folders.ADD_FOLDERS;
export const FETCH_FOLDERS = "mail/" + folders.FETCH_FOLDERS;
export const CREATE_FOLDER = "mail/" + folders.CREATE_FOLDER;
export const RENAME_FOLDER = "mail/" + folders.RENAME_FOLDER;
export const REMOVE_FOLDER = "mail/" + folders.REMOVE_FOLDER;
export const SET_UNREAD_COUNT = "mail/" + folders.SET_UNREAD_COUNT;
export const TOGGLE_FOLDER = "mail/" + folders.TOGGLE_FOLDER;

export const TOGGLE_EDIT_FOLDER = "mail/" + folderList.TOGGLE_EDIT_FOLDER;

export const ADD_MAILBOXES = "mail/" + mailboxes.ADD_FOLDER;
export const FETCH_MAILBOXES = "mail/" + mailboxes.FETCH_MAILBOXES;
