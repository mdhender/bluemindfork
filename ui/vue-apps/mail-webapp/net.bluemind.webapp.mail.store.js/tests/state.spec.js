import * as state from "../src/state";
import { DraftStatus } from "@bluemind/backend.mail.store";

describe("[Mail-WebappStore][state] : initial state", () => {
    test("contains an undefined currentFolderKey", () => {
        expect(state.currentFolderKey).toBeUndefined;
    });
    test("contains an undefined currentMessageKey", () => {
        expect(state.currentMessageKey).toBeUndefined;
    });
    test("contains an undefined login", () => {
        expect(state.login).toBeUndefined;
    });
    test("contains a 'draft'", () => {
        expect(state.draft).toEqual({
            id: null,
            status: DraftStatus.NEW,
            saveDate: null,
            attachments: []
        });
    });
    test("contains a empty object 'foldersData'", () => {
        expect(state.foldersData).toEqual({});
    });
    test("contains a 'search'", () => {
        expect(state.search).toEqual({
            pattern: null,
            loading: false,
            error: false
        });
    });
    test("contains a 'messageFilter'", () => {
        expect(state.messageFilter).toBeUndefined();
    });
    test("not to contain anything else", () => {
        expect(Object.keys(state).sort()).toEqual(
            [
                "currentFolderKey",
                "currentMessageKey",
                "currentMessageParts",
                "draft",
                "foldersData",
                "login",
                "search",
                "messageFilter"
            ].sort()
        );
    });
});
