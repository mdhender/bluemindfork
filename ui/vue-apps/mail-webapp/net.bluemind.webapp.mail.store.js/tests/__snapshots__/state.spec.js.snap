// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Mail-WebappStore][state] initial state 1`] = `
Object {
  "currentFolderKey": undefined,
  "foldersData": Object {},
  "login": undefined,
  "maxMessageSize": undefined,
  "messageFilter": undefined,
  "status": "idle",
}
`;
